### PR TITLE
introduce an optional ignoreFn option

### PR DESCRIPTION
**Pull request by gustavnikolaj**
and change the honorIndex boolean arg to an options object

This allows people to selectably ignore certain files. The problem
caused my build to fail as I had a dependency on a npm package which had
a file called the same as the package, and this plugin took precedence
over resolving through the main property in package.json.

This allows me to short circuit the plugin for certain requests.

```js
    new webpack.ResolverPlugin(new DirectoryNamedWebpackPlugin({
      ignoreFn: function (request) {
        return request.path.indexOf('node_modules') !== -1;
      }
    }))
```

### CODE REVIEW

**Comment by lucasmotta** [comment]
Hey @shaketbaby, any way you could look into this? I'm having some problems with some libraries inside `node_modules`, so it would be great to have this feature.

**Comment by shaketbaby** [comment]
I didn't get email notification for some reason. Will look at it.

File: index.js [hunk 1]
```diff
@@ -1,16 +1,23 @@
...
 module.exports = DirectoryNamedWebpackPlugin;
 
-function DirectoryNamedWebpackPlugin(honorIndex) {
-  this.honorIndex = !!honorIndex;
+function DirectoryNamedWebpackPlugin(options) {
+  options = options || {};
+  this.options = {
+    honorIndex: !!options.honorIndex,
+    ignoreFn: options.ignoreFn || function (x) { return false; }
+  };
```
Comment (shaketbaby) [inline index.js]: I am afraid this wil break existing code. Will have to check whether options is an Object or bool and initialize `this.options` accordingly. Do you mind to make it backward compatible?

**Comment by shaketbaby** [comment]
Hey @lucasmotta I've released a new version 1.2.0 to include this, hope not too late.

**Comment by gustavnikolaj** [comment]
Hey @shaketbaby. Thanks for merging! Sorry that I didn't have time to look at your proposed changes yet. But I can see that you took care of it yourself :+1:

**Pull request merged**